### PR TITLE
misc(docs): Update getting started docs

### PR DESCRIPTION
**Pull request by barbados-clemens**
Added a glossary page of common terms.
Updated nx core example
updated getting started docs

### CODE REVIEW

File: docs/shared/getting-started/nx-setup.md [hunk 1]
```diff
@@ -2,14 +2,16 @@
...
-Run the following command to create a new workspace.
+Run the following command to create a new [workspace](/getting-started/glossary#workspace).
 
 ```bash
 # pass @latest in case npx cached an older version of create-nx-workspace
 npx create-nx-workspace@latest
 ```
 
-When creating a workspace, you will have to choose a preset, which will preconfigure a few things for you.
+When creating a workspace, you will have to choose a preset, which will preconfigure a few things for you. If you're unsure se
```
Comment (isaacplmann) [inline docs/shared/getting-started/nx-setup.md:12]: looks like you didn't finish your sentence here

File: docs/shared/getting-started/nx-setup.md [hunk 2]
```diff
@@ -28,11 +30,12 @@ Some presets set up applications, e2e tests, etc.
 npx create-nx-workspace --preset=react
 npx create-nx-workspace --preset=react-native
 npx create-nx-workspace --preset=angular
+npx create-nx-workspace --preset=@nrwl/remix
```
Comment (isaacplmann) [inline docs/shared/getting-started/nx-setup.md:33]: is this released?

File: docs/shared/getting-started/nx-setup.md [hunk 3]
```diff
@@ -44,7 +47,7 @@ If you have an existing Create React App project, you can gain the benefits of N
 npx cra-to-nx
 ```
 
-For more information on adding Nx to an existing repository see the [migration guide](/migration/migration-cra)
+Learn more about migrating Create React Apps to Nx via our [Create React App migration guide](/migration/migration-cra)
```
Comment (isaacplmann) [inline docs/shared/getting-started/nx-setup.md:50]:
```suggestion
Learn more about migrating Create React App projects to Nx via our [Create React App migration guide](/migration/migration-cra)
```

File: docs/shared/glossary.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
+# Glossary
+
+## Computational Cache
+
+A mechanism that allows Nx to understand the changes made to the workspace.
+
+[Learn more](/using-nx/caching)
+
+## Dependency Graph (dep-graph)
```
Comment (isaacplmann) [inline docs/shared/glossary.md:9]:
```suggestion
## Project Graph
```

Comment (isaacplmann) [reply docs/shared/glossary.md:9]: We've renamed it elsewhere in the docs

File: docs/shared/glossary.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+A representation of all the dependencies between projects in the workspace.
+
+See a visual representation of the dependency graph by running `npx nx dep-graph` or check out
+this [demo video](https://www.youtube.com/watch?v=UTB5dOJF43o).
+
+## Distributed Task Execution (DTE)
+
+A feature of Nx that allows the execution of a task graph across multiple machines instead of locally.
```
Comment (isaacplmann) [inline docs/shared/glossary.md:18]:
```suggestion
Nx Cloud coordinates executing tasks across multiple machines, balancing out the work load according to the time each task takes to complete.
```

File: docs/shared/glossary.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+[Learn more](/using-nx/dte)
+
+## Generators
+
+Generators provide a way to automate many tasks you regularly perform as part of your development workflow.
```
Comment (isaacplmann) [inline docs/shared/glossary.md:24]:
```suggestion
Generators automate adding, modifying or deleting code.
```

File: docs/shared/glossary.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+A way to extend the functionality of Nx. Plugins help in various ways such as automating boilerplate tasks and
+streamlining integrations It's important to note that plugins are not required to use Nx. Learn more
+about [Nx without plugins](/getting-started/nx-core)
+
+[Learn more about Nx plugins](nx-plugin/overview)
+
+## Tags
+
+a generic mechanism for expressing constraints
```
Comment (isaacplmann) [inline docs/shared/glossary.md:58]:
```suggestion
A generic mechanism for expressing constraints between projects.
```

File: docs/shared/glossary.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+## Monorepo
+
+A monorepo is a single repository containing multiple distinct projects, with well-defined relationships.
+
+[Learn more](https://monorepos.tools)
+
+## Plugin
+
+A way to extend the functionality of Nx. Plugins help in various ways such as automating boilerplate tasks and
+streamlining integrations It's important to note that plugins are not required to use Nx. Learn more
```
Comment (isaacplmann) [inline docs/shared/glossary.md:51]:
```suggestion
streamlining integrations. It's important to note that plugins are not required to use Nx. Learn more
```

**Review by isaacplmann — COMMENTED** [review]
More than half of the words in the glossary aren't actually linked from anywhere.  Also, almost all the words in the glossary have a full page dedicated to explaining them.  Would it be better to link directly to that full page whenever the word is mentioned?

**Comment by barbados-clemens** [comment]
> More than half of the words in the glossary aren't actually linked from anywhere. Also, almost all the words in the glossary have a full page dedicated to explaining them. Would it be better to link directly to that full-page whenever the word is mentioned?

I'm conflicted as I like the idea of having all these terms quickly accessible on a single page with simple straight forward definitions, but the full page for some terms is very handy for a deep dive, which we link to I'm not too sure about (assuming we keep both)

I feel like within a guide linking to the full page as the idea is "if you wanna go deeper you can read these full pages" but from a place that just mentions the term without explaining a link to the glossary. 
BUT this (potentially?) creates a fragmented experience when someone is wanting to learn about a term, as it maybe is a full guide/deep dive or a simple definition. 

I think _ideally_ we could have a tooltip/"what's this" type UI that will display the quick definitions from the glossary, and if a full-page guide exists have a link out to "go deeper" on the topic. but that might require some extra leg work that's probably out of the scope of this PR. (assuming we want to do that)

**Comment by isaacplmann** [comment]
I'm conflicted too.  Let's open up the question to more people and see what others think.  I'll link this PR in the nx channel.

**Review by johannesschobel — CHANGES_REQUESTED** [review]
a lot of duplicate text.. can't we just somehow "link" those texts? i.e., move the texts to a library and include them in the different docs?

**Pull request closed without merging**